src/gtk/textctrl.cpp

```diff
@@ -61,6 +61,37 @@ static void wxGtkOnRemoveTag(GtkTextBuffer *buffer,
 }
 }
 
+extern "C"
+{
+static void maxlen_handler(GtkTextBuffer *buffer, GtkTextIter *location, gchar *WXUNUSED(text), gint len, wxTextCtrl *win)
+{
+    auto count = gtk_text_buffer_get_char_count( buffer );
+    auto maxlen = win->GetMaxLength();
+    if( count > maxlen )
+    {
+        GtkTextIter offset, end;
+        auto startPos = gtk_text_iter_get_offset( location );
+        gtk_text_buffer_get_iter_at_offset( buffer, &offset, startPos - 1 );
+        gtk_text_buffer_get_iter_at_offset( buffer, &end, ( startPos - 1 ) + len );
+        // This function is connected using g_signal_connect_after() call, therefore
+        // direct buffer modification is required.
+        // Using g_signal_connect() doesn't work as the text is still being entered
+        // probably because the signal is documented as "Run Last". For exactly same
+        // reason, ("Run Last") it won't work in GTKOnInsertText() as it called from
+        // the handler that does not connected after
+        gtk_text_buffer_delete( buffer, &offset, &end );
```

Comment:
Ideally we should delete just enough to stay under `maxlen` and not all new entry.